jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -117,15 +117,17 @@ ecma_builtin_helper_object_to_string_tag_helper (ecma_value_t tag_value) /**< st
 ecma_value_t
 ecma_builtin_helper_object_to_string (const ecma_value_t this_arg) /**< this argument */
 {
-  lit_magic_string_id_t type_string;
+  ecma_stringbuilder_t builder = ecma_stringbuilder_create_raw ((lit_utf8_byte_t *) "[object ", 8);
```

Comment:
This could be delayed right until the end, that way you don't need to destroy it on errors.